packages/fhir-care-team/src/components/ListView/index.tsx

```diff
@@ -119,18 +119,15 @@ export const CareTeamList: React.FC<CareTeamListPropTypes> = (props: CareTeamLis
   const { currentPage, pageSize } = pageProps;
   const pageOffset = (currentPage - 1) * pageSize;
 
-  const { data, isLoading, isFetching, error, refetch } = useCareTeamsHook(
+  const { data, isError, error, refetch, isSuccess } = useCareTeamsHook(
     fhirBaseURL,
     pageSize as number,
     pageOffset,
     setPayloadCount
   );
+  if (isError) return <BrokenPage errorMessage={`${error}`} />;
 
-  if (isLoading || isFetching) return <Spin size="large" />;
-
-  if (error) {
-    return <BrokenPage errorMessage={`${error}`} />;
-  }
+  if (!isSuccess) return <Spin size="large" />;
```

Comment:
@ashfahan due to react query's default [aggresive refetching nature](https://tkdodo.eu/blog/status-checks-in-react-query) I would suggest we structure the loader status rendering as follows:
```
  // isLoading is only true during the first fetch
  // we probably do not want to show a hard spinner during re-fetches
  if (isLoading) return <Spin size="large" />;

  // first check that we have cached data, and use the stale data to render the page
  if (data) {
    // render the page here
  }

  // handle errors last -> this means should a refetch fail but we have data
  // the user can still interact with the stale data

  if (error) {
    return <BrokenPage errorMessage={`${error}`} />;
  }
```
The refetch's are not handled in this scenario.